### ISSUE TITLE
Circular dependency - web, cli, browser

### ISSUE THREAD

**Issue by frankcarey** [blinkreaction/drude#43]
When uncommenting the seleniume2/browser containers in the docker-compose.yml file, docker complains about the circular dependency.  Here is a copy of the file.

``` Yaml
# Docker and Docker Compose based environment for Drupal.
# See https://github.com/blinkreaction/docker-fig-drupal for more information and documentation.

# Web node
web:
  hostname: web
  image: blinkreaction/drupal-apache:2.2
  ports:
    - "80"
    - "443"
    # Use IP based mapping when working with multiple projects. See docs (^^) for more details.
    #- 192.168.10.10:80:80 
    #- 192.168.10.10:443:443
  environment:
    - VIRTUAL_HOST=devinci.local
  volumes_from:
    - cli
  links:
    - cli
    - db
    # Uncomment this and the memcached service definition below to start using memcached.
    #- memcached
  environment:
    - VIRTUAL_HOST=hello-world.drude

# DB node
db:
  hostname: db
  image: blinkreaction/drupal-mysql:5.5
  ports:
    - "3306"
    # Use IP based mapping when working with multiple projects. See docs (^^) for more details.
    #- 192.168.10.10:3306:3306
  # DB configuration
  environment:
    - MYSQL_ROOT_PASSWORD=admin123
    - MYSQL_USER=drupal
    - MYSQL_PASSWORD=123
    - MYSQL_DATABASE=drupal
  volumes:
     # MySQL configuration overrides
     - "./.drude/etc/mysql/my.cnf:/etc/mysql/conf.d/z_my.cnf"

# CLI node
# Used for all console commands and tools.
cli:
  hostname: cli
  image: blinkreaction/drupal-cli:php-5.6
  volumes:
    # Host SSH keys mappings.
    # We try to map both options. The startup.sh script in cli container will decide which one to use.
    - /.ssh:/.ssh-b2d  # boot2docker-vagrant
    - ~/.ssh:/.ssh   # Linux
    # PHP configuration overrides
    - "./.drude/etc/php5/php.ini:/etc/php5/fpm/conf.d/z_php.ini"
    #- "./.drude/etc/php5/php-cli.ini:/etc/php5/fpm/conf.d/z_php.ini"
    # Project root folder mapping
    - ".:/var/www"
  links:
    - db
    # Uncomment this and the browser service definition below to start using selenium2.
    - browser

# Memcached node
# Uncomment the service definition section below and the link in the web service above to start using memcached.
# memcached:
#   hostname: memcached
#   image: memcached
#   environment:
#     # Memcached memory limit in megabytes
#     - MEMCACHED_MEMORY_LIMIT=128

# ngrok node
# Uncomment the service definition section below to start using ngrok for sharing your local web server with the world.
# share:
#   hostname: share
#   image: fnichol/ngrok
#   ports:
#     - "4040"
#     # Use IP based mapping when working with multiple projects. See docs (^^) for more details.
#     #- 192.168.10.10:4040:4040
#   links:
#     - web:http

# selenium2 node
# Uncomment the service definition section below and the link in the web service above to start using selenium2 driver for Behat tests requiring JS support.
browser:
   hostname: browser
   image: selenium/standalone-chrome
   links:
     - web
   ports:
     - "4444"
```

**Comment by lmakarov** [comment]
Yeah, that's an issue I ran into last month while refactoring the setup - having PHP in a single place (cli container) instead of two places (web and cli).
I don't have a solution for this yet and this blocks the release of the refactored configuration.

**Comment by joshbenner** [comment]
Why does web depend on cli?

**Comment by lmakarov** [comment]
@joshbenner web = apache (only), cli = php (both php-fcgi and php-cli)
The goal is to have a single PHP in one place (cli) instead of two (web and cli), so there are no version mismatches, less maintenance overhead, etc.

**Comment by joshbenner** [comment]
IMO: I'd avoid coupling web and cli like that, and instead move the responsibility of version matching to the use of tags in docker-compose.yml.

**Comment by lmakarov** [comment]
@joshbenner having PHP in two places in redundant and produces many webserver-php version combinations that need to be maintained. What kind of issues (besides this one) do you see in moving php-fcgi service out of the web container into cli?

**Comment by joshbenner** [comment]
Are you running PHP as a separate process (ie: FPM), or are you using an Apache module? If you are using a module, then PHP is part of the apache process and should be in the same container. If you are using a separate process, then having separate containers is acceptable, but I would keep that container distinct from the "cli" container. This suggests you'd have a "php" image, which two containers (php and cli) would both use. This would also let you avoid the dependency issues.

**Comment by lmakarov** [comment]
Fixed in [v1.0.0-rc1](https://github.com/blinkreaction/drude/releases/tag/v1.0.0-rc1)
Moved the new experimental setup into a feature branch: https://github.com/blinkreaction/drude/tree/feature/no-php-in-web
